api/disruption_subspec.go

```diff
@@ -11,9 +11,13 @@ import (
 	chaostypes "github.com/DataDog/chaos-controller/types"
 )
 
-// DisruptionArgsGenerator generates args for the given disruption
-type DisruptionArgsGenerator interface {
+// Contains all methods required for a disruption sub-specification (Network, DNS, CPUPressure, etc.)
+type DisruptionSubSpec interface {
```

Comment:
are subspec and "kind" the same thing?

Reply:
Yes, but `DisruptionKind` as a type name is already taken. Not to bikeshed, whatever you want to name it is fine, but I'll put my 2 cents in for `DisruptionType` or `IDisruptionKind`. I'll even put in 4 cents for renaming `types.DisruptionKind` to `types.DisruptionKindName` and then call this type `DisruptionKind`

Reply:
Done ! It does feel good once changed, that was the good call
Revamping the tests today and we can push this through :)